include/circt/Dialect/Pipeline/PipelineOps.td

```diff
@@ -148,19 +147,45 @@ def ScheduledPipelineOp : PipelineBase<"scheduled", [
 
     Any value defined outside the pipeline is considered an external input. An
     external input will _not_ be registered.
+
+    The pipeline may optionally be provided with an array of bits `stallability`
+    which is used to determine which stages are stallable.
+    - If not provided and the pipeline has a stall signal, all stages are stallable.
+    - If provided, and the pipeline has a stall signal, the number of bits must
+      match the number of stages in the pipeline. Each bit represents a stage,
+      in the order of which the stages appear wrt. the `pipeline.stage` operations.
+      A bit set to 1 indicates that the stage is stallable, and 0 indicates that
+      the stage is not stallable.
+    A non-stallable stage will not have a stall signal connected to its
+    pipeline registers.
+    The exit stage of a pipeline is never stallable.
```

Comment:
If the exit stage writes to a FIFO, and that FIFO can go full arbitrarily, then the exit stage needs to be stallable to avoid fifo overflow.  I was expecting the dual restriction: the entry stage must always be stallable.  Or are you requiring the user of the pipeline to avoid starting a new token down the pipeline when the top level stall signal is high?

Reply:
The "exit" stage is a bit of a funky concept given that it's non-registered, hence as far as i understand, there really isn't anything to stall in the exit stage.
i.e.:
```mlir
pipeline.scheduled(...) {
    ...
    // This is the stage prior to the exit stage. Will infer registers.
    pipeline.stage ^bb_exit(...)
  ^bb_exit(%exit_enable : i1):
    // this is the exit stages
    // pipeline.return doesn't infer registers
    pipeline.return
}
```
In practice, all pipelines that we emit for Ibis will most likely not have _any_ logic in the exit stage. 

> are you requiring the user of the pipeline to avoid starting a new token down the pipeline when the top level stall signal is high?

Yes. Pipelines in the `pipeline` dialect has no way of backpressuring, hence it's the users responsibility to ensure that the driver for the top-level `stall` signal will also be factored into backpressuring the top-level `go` signal.